### PR TITLE
SC-3307 configurable 401 redirect

### PR DESCRIPTION
**Pull request by Parnswir**
# Issue: [SC-3307](https://ticketsystem.schul-cloud.org/browse/SC-3307)

[legacy client PR](https://github.com/schul-cloud/schulcloud-client/pull/1615)

## Description

- Redirects unauthenticated users back to the requested page after login.
- Adds an optional environment variable to send users to a different page than `/login` (mainly for THR): `NOT_AUTHENTICATED_REDIRECT_URL` (see [legacy client PR](https://github.com/schul-cloud/schulcloud-client/pull/1615))
- Fixes two error messages during logout (due to not having a user object at hand)

## Screenshot

❌

### CODE REVIEW

File: src/layouts/legacyLoggedIn.vue [hunk 1]
```diff
@@ -133,9 +133,11 @@ export default {
 		sidebarItems() {
 			const sidebarItems = this.sidebarBaseItems.filter((item) => {
 				const hasRequiredPermission =
+					this.user &&
 					this.user.permissions &&
 					this.user.permissions.includes(item.permission);
```
Comment (adrianjost) [inline src/layouts/legacyLoggedIn.vue]: I think you can use optional chaining in this repo 🎉 At least I hope so and did it somewhere else. 
```suggestion
hasRequiredPermission =
					this.user?.permissions &&
					this.user?.permissions.includes(item.permission);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/layouts/legacyLoggedIn.vue [hunk 2]
```diff
@@ -133,9 +133,11 @@ export default {
...
 				const hasExcludedPermission =
+					this.user &&
 					this.user.permissions &&
```
Comment (adrianjost) [inline src/layouts/legacyLoggedIn.vue]: here as well

File: src/layouts/legacyLoggedIn.vue [hunk 3]
```diff
@@ -133,11 +133,9 @@ export default {
 		sidebarItems() {
 			const sidebarItems = this.sidebarBaseItems.filter((item) => {
 				const hasRequiredPermission =
-					this.user.permissions &&
-					this.user.permissions.includes(item.permission);
+					this.user?.permissions?.includes(item.permission);
 				const hasExcludedPermission =
-					this.user.permissions &&
-					this.user.permissions.includes(item.excludedPermission);
+					this.user?.permissions?.includes(item.excludedPermission);
```
Comment (adrianjost) [inline src/layouts/legacyLoggedIn.vue]:
```suggestion
					this.user?.permissions && this.use?.permissions.includes(item.excludedPermission);
```

File: src/layouts/legacyLoggedIn.vue [hunk 4]
```diff
@@ -133,11 +133,9 @@ export default {
 		sidebarItems() {
 			const sidebarItems = this.sidebarBaseItems.filter((item) => {
 				const hasRequiredPermission =
-					this.user.permissions &&
-					this.user.permissions.includes(item.permission);
+					this.user?.permissions?.includes(item.permission);
```
Comment (adrianjost) [inline src/layouts/legacyLoggedIn.vue]:
```suggestion
					this.user?.permissions && this.user.permissions.includes(item.permission);
```

File: src/components/legacy/TheFooter.vue [hunk 5]
```diff
@@ -65,7 +65,7 @@ export default {
 							text: "HPI",
 						},
 						{
-							href: defaultDocuments.specificFiles(this.school.documentBaseDir)
+							href: defaultDocuments.specificFiles(this.school?.documentBaseDir)
```
Comment (adrianjost) [inline src/components/legacy/TheFooter.vue]:
```suggestion
							href: defaultDocuments.specificFiles(this.school?.documentBaseDir || "default")
```
sonst funktioniert im Footer der Muster Schulen Link in ausgeloggtem Zustand nicht.

Comment (dominikbruechner) [reply src/components/legacy/TheFooter.vue]: Außerdem, wird das auch in den themed footers benötigt? evtl muss es noch an den anderen stellen nachgezogen werden

Comment (adrianjost) [reply src/components/legacy/TheFooter.vue]: pushed a better fix that fixes this directly in the method.

**Pull request merged**